pyccel/parser/base.py

```diff
@@ -500,17 +503,118 @@ def insert_import(self, expr):
                         container[source] = []
                     container[source] += name
 
-    def print_namespace(self):
+    def dump(self, filename=None):
+        """
+        Dump the current ast using Pickle.
 
-        # TODO improve spacing
+          Parameters
+          ----------
+          filename: str
+            output file name. if not given `name.pyccel` will be used and placed
+            in the Pyccel directory ($HOME/.pyccel)
+        """
+        import pickle
+        import hashlib
+
+        if not filename:
+            if not self.filename:
+                raise ValueError('Expecting a filename to load the ast')
+
+            path , name  = os.path.split(self.filename)
+
+            if not name.split('.')[-1] == 'pyh':
+                return
+            else:
+                name = name[:-4]
 
-        print ('------- namespace -------')
-        for (k, v) in self.namespace.items():
-            print ('{var} \t :: \t {dtype}'.format(var=k, dtype=type(v)))
-        print ('-------------------------')
+            name     = '{}.pyccel'.format(name)
+            filename = path+ '/' + name
```

Comment:
Here we should use `os.path.join`, which is shell-agnostic

Reply:
done